src/map/unit.cpp

```diff
@@ -543,7 +541,8 @@ static TIMER_FUNC(unit_walktoxy_timer)
 			// Walk skills are triggered regardless of target due to the idle-walk mob state.
 			// But avoid triggering on stop-walk calls.
 			if(!ud->state.force_walk && tid != INVALID_TIMER &&
-				!(ud->walk_count%WALK_SKILL_INTERVAL) &&
+				DIFF_TICK(tick, md->last_skillcheck) > MOB_SKILL_INTERVAL - md->status.speed / 2 &&
```

Comment:
This is to get as close to 1 second as possible. e.g. if movement speed is 300 we don't want the monsters to use a skill every 1000-1300ms, but rather every 850-1150ms, so that the average is 1000ms.